junit5/src/main/java/org/jboss/weld/junit5/auto/SetSyntheticArchiveDiscoveryMode.java

```diff
@@ -0,0 +1,26 @@
+package org.jboss.weld.junit5.auto;
+
+import org.jboss.weld.bootstrap.spi.BeanDiscoveryMode;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Sets discovery mode for Weld SE synthetic bean archive. Valid options are {@link BeanDiscoveryMode#ALL} and
+ * {@link BeanDiscoveryMode#ANNOTATED}.
+ * <p>
+ * Starting with Weld 5 (CDI 4), the default value is {@code ANNOTATED}. Applications can leverage this annotation to
+ * enforce same behavior as older Weld versions where synthetic archives used discovery mode {@code ALL}.
+ * <p>
+ * This annotation is equal to invocation of {@link org.jboss.weld.environment.se.Weld#setBeanDiscoveryMode(BeanDiscoveryMode)}.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@Inherited
+public @interface SetSyntheticArchiveDiscoveryMode {
```

Comment:
How about `@SetBeanDiscoveryMode`? The "SyntheticArchive" part would not be very helpful for most users anyway...

Reply:
Hmm, you may be right, after all, when you use the annotation approach, we are automatically feeding all classes to the synthetic archive anyway and there should be no standard discovery happening.

Reply:
Yes, and the`SetSyntheticArchiveDiscoveryMode` seems to be tooooo long ;-)

Reply:
> Yes, and the`SetSyntheticArchiveDiscoveryMode` seems to be tooooo long ;-)

Riiiight, so `@SSADM`? xD

Jokes aside, I have changed to name to what you suggested @mkouba.
I've kept the javadoc mentions of the synthetic archive for completeness sake.